source/extensions/certificate_providers/certificate_provider.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+
+#include "openssl/ssl.h"
+#include "openssl/x509v3.h"
+
+
+namespace Envoy {
+namespace Extensions {
+namespace CertificateProviders {
+
+struct Certpair {
+    X509 cert_;
+    EVP_PKEY key_;
```

Comment:
We don't want to leak OpenSSL details into headers, consider use Envoy::Common::Crypto::CryptoObject instead.